assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Overview.js

```diff
@@ -207,9 +221,12 @@ const Overview = ( {
 				{ ( ! analyticsModuleConnected || ! analyticsModuleActive ) &&
 					! isNavigatingToReauthURL && (
 						<Cell { ...halfCellProps }>
-							{ ! analyticsModuleActive && (
-								<ActivateModuleCTA moduleSlug="analytics" />
-							) }
+							{ ! analyticsModuleActive &&
+								( zeroDataStatesEnabled ? (
+									<ActivateAnalyticsCTA />
+								) : (
+									<ActivateModuleCTA moduleSlug="analytics" />
+								) ) }
```

Comment:
To be honest, I didn't think this was any worse than the level of nesting elsewhere in the codebase. However, that's not a good thing per-se - and I actually do think we would benefit from introducing a [cyclomatic complexity](https://eslint.org/docs/rules/complexity) lint rule to warn and reduce about levels of nesting (maybe we already do have one but it's quite relaxed, I've not seen it yet anyway).

Happy to make the change you've suggested. :+1:

Reply:
As I've mentioned it, I figured I might as well create an issue to suggest the introduction of the complexity rule: https://github.com/google/site-kit-wp/issues/4895

Reply:
On reflection I have realised that my thinking was also along the lines of this complexity would be reduced once the feature flag is removed. No reason not to make the change you've suggested, though!

Reply:
Done :+1:

Reply:
> Also I am a bit of a fan of the semantic value in a ternary - it indicates the `if`/`else`, mutual exclusivity at a glance

That's true, though I tend to find it a much terser syntax than actual if/else. I long for Ruby's inline `foo = true if bar else false` syntax.

Anyway, I find ternaries specifically in JSX to get really tough-to-follow and trace to their end, because of all of the parenthesis involved. Semantically/syntactically it's better, 100%. But legibility I find it to be so difficult it to parse through multi-line ternaries that the _semantic correctness_ of it is nullified by the illegibility of it.

My generally thinking is that multi-line ternaries are to be avoided in C-style languages for legibility, and once JSX is involved I find it even worse. It's certainly a preference, but—anecdotally—I've spoken with a lot of designers and junior developers who find ternaries in "template" code like PHP, JSX, etc. to be very difficult to follow and understand. So I tend to stray away from it.

In our case it might well mean we need more sub-components or sections of the JSX defined before the final `return ()` call. 🤔